cmd/root/root.go

```diff
@@ -50,6 +50,25 @@ func GetCommand() *cobra.Command {
 			if err := config.PersistentStartupProcessFlags(); err != nil {
 				return err
 			}
+			pidFilePath := viper.GetString(config.OptPIDFile)
+			pidFile, err := cli.NewPIDFile(pidFilePath)
+			if err != nil {
+				return err
+			}
+			err = pidFile.Acquire()
+			if err != nil {
+				return err
+			}
+			ctx := context.WithValue(cmd.Context(), config.OptPIDFile, pidFile)
```

Comment:
I don't love using this constant for both the viper name and the context key.

also I think context keys should have user-defined types, not be strings?

Reply:
actually I think I'd almost rather store this in a global variable at the top of this file, it feels like a lesser sin